### DISCUSSION TITLE
AI powered features

### DISCUSSION THREAD

**Discussion by zewa666** [dineug/erd-editor#380] [Show and tell]
I was playing around with utilizing AI to support the user in creating table columns and here is a rough shot of what it potentially could look like:

![ai-gen-columns](https://github.com/user-attachments/assets/b7d2d6d1-d7b9-4307-93fa-938a0f7b5f20)

note that I'm running locally on a Phi3 with a super old desktop PC :)

```js
[{
  role: 'system',
  content: `You will receive a json representation of a ERD SQL Table for MSSQL. Your job is to suggest up to 3 additional columns.
the format of the json is as follows: { tableName: string, comment: string, columns: [{ dataType: string, name: string, default: any }]}.
Please reply only with a valid json, e.g do not use single quotes but always double quotes, response and no other text.`,
},
{
  role: 'user',
  content: JSON.stringify({ tableName: table.name, comment: table.comment, columns }),
}]
```

**Comment by dineug** [comment]
Wow, this is very impressive!
It's a great starting point.

**Reply by zewa666** [reply]
Yup thats right. So much opportunities to do something helpful with LLMs to cut boilerplate.

Here's one more use-case ... generate a related table.

![ai-gen-related-table](https://github.com/user-attachments/assets/9895ab46-84f5-4855-8769-23f1a4c91747)

This one would ideally branch away and ask the user whether you want a "child" or "parent" table. E.g the generated Manufacturer would act as a primary key and add a FK to the Engine table + draw a One-to-Many relation.

Btw I very appreciate the homegrown state management solution. super straight forward and makes re-using exsting behavior quite simple

**Comment by dineug** [comment]
#452
